CMakeLists.txt

```diff
@@ -12,8 +14,16 @@ set(ONNX_ROOT ${PROJECT_SOURCE_DIR})
 set(CMAKE_MODULE_PATH "")
 list(APPEND CMAKE_MODULE_PATH ${ONNX_ROOT}/cmake/Modules)
 
-if(TARGET protobuf::protoc)
-  set(PROTOBUF_PROTOC_EXECUTABLE $<TARGET_FILE:protobuf::protoc>)
+if(TARGET protobuf::libprotobuf)
+  # Sometimes we need to use protoc compiled for host architecture while
+  # linking libprotobuf against target architecture. See
+  # https://github.com/caffe2/caffe2/blob/96f35ad75480b25c1a23d6e9e97bccae9f7a7f9c/cmake/ProtoBuf.cmake#L92-L99
+  if(EXISTS "${ONNX_CUSTOM_PROTOC_EXECUTABLE}")
```

Comment:
Maybe use `if(DEFINED ONNX_CUSTOM_PROTOC_EXECUTABLE)` instead? If `-DONNX_CUSTOM_PROTOC_EXECUTABLE` is specified, but doesn't exist (due to user error), the outcome would be very inintuitive: CMake will try to build and use the `protoc` for a different platform

Reply:
For this usecase, we would like to use `ONNX_CUSTOM_PROTOC_EXECUTABLE` as a truth. Actually, if we fallback to let cmake build its own protoc, it can fail and error message will not be as intuitive (Check Caffe2 CI for Android and iOS).